src/ZkFold/Symbolic/Algorithms/Hash/SHA2.hs

```diff
@@ -227,30 +221,25 @@ instance (KnownNat n, FromConstant Natural (ByteString n b)) => ToWords Natural
 
 -- | Constraints required for a SHA2 of a Natural number.
 --
-type SHA2N algorithm backend =
-   ( AlgorithmSetup algorithm backend
-   , NFData (ByteString (WordSize algorithm) backend)
-   , FromConstant Natural (ByteString (ChunkSize algorithm) backend)
-   , KnownNat (ChunkSize algorithm)
-   , KnownNat (WordSize algorithm)
-   , Iso (UInt (WordSize algorithm) Auto backend) (ByteString (WordSize algorithm) backend)
-   , Iso (ByteString (WordSize algorithm) backend) (UInt (WordSize algorithm) Auto backend)
-   , AdditiveSemigroup (UInt (WordSize algorithm) Auto backend)
-   , BoolType (ByteString (WordSize algorithm) backend)
-   , ShiftBits (ByteString (WordSize algorithm) backend)
-   , ToWords (ByteString (ChunkSize algorithm) backend) (ByteString (WordSize algorithm) backend)
-   , Concat (ByteString (WordSize algorithm) backend) (ByteString (8 * WordSize algorithm) backend)
-   )
+type SHA2N algorithm context =
```

Comment:
Maybe we can move all of these to the superclass constraints of `AlgorithmSetup`? This way, `type SHA2` would also become shorter